docs/book/src/tasks/experimental-features/ignition.md

```diff
@@ -0,0 +1,8 @@
+# Experimental Feature: Ignition Bootstrap Config (alpha)
+
+The default configuration engine for bootstrapping machines is cloud-init. To bootstrap machines
+using Ignition instead, enable this feature.
```

Comment:
can we please add a caveat section here clarifying this is only used/tested/supported? by flatcar container linux and that is tied to ign version 2 atm?

Reply:
Sounds good to me :+1:

Reply:
Also please let's be specific on the PR title and eventual final commit, e.g `Add Ignition-based Flatcar Container Linux support to CABPK.`

Reply:
>can we please add a caveat section here clarifying this is only used/tested/supported? by flatcar container linux and that is tied to ign spec version 2 atm?

Yes, good point. I'm currently editing the docs so will add that, too.

Reply:
>Also please let's be specific on the PR title and eventual final commit, e.g Add Ignition-based Flatcar Container Linux support to CABPK.

I think it's accurate to say we're adding generic Ignition v2 support using Flatcar for testing. I don't see a reason other distros which use Ignition v2 shouldn't work as we aren't tailoring the feature for Flatcar only (right?).

Reply:
As far as I know, there are no other distros using Ignition spec 2.  Since that spec is no longer supported upstream, any distros adding Ignition support should implement spec 3 instead.  So this support is essentially only for Flatcar and any future downstreams of it.

Also worth noting that the feature-set of Container Linux Configs, and the CLC Transpiler, was designed to be specific to Container Linux.  Any distro that wanted to support spec 2 and wasn't from the Container Linux lineage was expected to implement their own transpiler.  (That policy has changed with Ignition spec 3: Butane is designed to support distinct config "variants" for different distros.)  So, as a practical matter, the spec 2 support added here is only useful for a Flatcar-like distro.

Reply:
Fair enough. How about we title the PR `Add support for generating Ignition v2 bootstrap data to CABPK to enable Flatcar Container Linux provisioning` @enxebre? If it's too long, we could also mention the Flatcar bit in the description.

In any case, FWIW I'm adding a caveat in the docs I'm currently writing which clearly states that right now the Ignition support has been tested with Flatcar only.

Reply:
>can we please add a caveat section here clarifying this is only used/tested/supported? by flatcar container linux and that is tied to ign spec version 2 atm?

@enxebre done:

- https://github.com/kubernetes-sigs/cluster-api/blob/66eb1bd608e0b217307a090bf239a9e46c446b99/docs/book/src/tasks/experimental-features/ignition.md#note
- https://github.com/kubernetes-sigs/cluster-api/blob/66eb1bd608e0b217307a090bf239a9e46c446b99/docs/book/src/tasks/experimental-features/ignition.md#caveats